bchehraz/react-challenge-project-jan-2022
# Challenge 7 - Update Dependencies

## Changes
_List Changes Introduced by this PR_
1. Updated all package dependencies via npm.
2. Updated `appRouter.js` and `login.js` which needed to be changed after `react-router-dom` was updated from v5 to v6.

## Purpose
_Describe the problem or feature in addition to a link to the issues._

- Link to issue: https://github.com/Shift3/react-challenge-project-jan-2022/issues/7
- The goal was to update all dependencies which will help to address the large number of security vulnerabilities present in the project. 

## Approach
_How does this change address the problem?_

Through the following approach, I was able to rid of all vulnerabilities, except for 2 moderate-level vulnerabilities found in the latest version of `react-scripts` that require manual fixes, according to `npm audit`. 

1. First I would run `npm outdated`, attempting to only update to the `wanted` version. However, there were still a large number of dependencies with vulnerabilities.
2. I then made the decision to just update the packages to their latest versions. I would do this one at a time, while rerunning the project after each update, in case any update introduced breaking changes. Upon further examination, the latest `react-router-dom` v5 to v6 now required changes to be made in both `appRouter.js` and `login.js`.
3. Proceeded to resolve the issues in `appRouter.js` and `login.js`.
4. Verified everything was done by running through the testing steps listed below.



## Pre-Testing TODOs
_What needs to be done before testing_
- The updates above, along with the changes to `appRouter.js` and `login.js` must be made prior to testing.

## Testing Steps
_How do the users test this change?_

1. Verified project runs as expected and displays no client-side errors on browser console.
2. Verified routes are still working by navigating pages.
3. Ran `npm test` to ensure any existing tests still pass.

## Learning
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

- I referred to the documentation in react-router to verify the changes I needed to make: https://reactrouterdotcom.fly.dev/docs/en/v6/getting-started/concepts#defining-routes

Closes https://github.com/Shift3/react-challenge-project-jan-2022/issues/7
